store/v2/root/factory.go

```diff
@@ -127,20 +127,30 @@ func CreateRootStore(opts *FactoryOptions) (store.RootStore, error) {
 		}
 	}
 
-	trees := make(map[string]commitment.Tree)
-	for _, key := range opts.StoreKeys {
+	newTreeFn := func(key string) (commitment.Tree, error) {
 		if internal.IsMemoryStoreKey(key) {
-			trees[key] = mem.New()
+			return mem.New(), nil
 		} else {
 			switch storeOpts.SCType {
```

Comment:
`storeOpts.SCType` appears to be the only value of importance bound within this closure when passed to `NewCommitStore`.  If `mountTreeFn` is strictly required for that constructor I suggest passing the SC type instead since this iteration of store/v2 only supports one type of SC tree anyway.